app/lib/payment_gateways/stripe_crypt.rb

```diff
@@ -44,13 +65,12 @@ def find_or_create_customer
       customer_id = payment_detail.credit_card_auth_code
       return create_customer if customer_id.blank?
 
-      begin
-        customer = Stripe::Customer.retrieve(customer_id, api_key)
-        return create_customer if customer.deleted?
-        customer
-      rescue Stripe::InvalidRequestError
-        create_customer
-      end
+      customer = Stripe::Customer.retrieve(customer_id, api_key)
+      return create_customer if customer&.deleted?
+
+      customer
+    rescue Stripe::InvalidRequestError
+      create_customer
```

Comment:
this changes logic. First of all I don't understand why do we rescue with `create_customer` with `create_customer`. Maybe retry? But then in the past, we only retried one of the options... just it is a little weird to read. 


As I see it, first create customer in case we don't find an ID. We retrieve customer by ID and create it in case it's status is deleted. And we rescue when in case the `retrieve` call or the `deleted?` call raised. It will retry if `create_customer` failed though.

I mean, I'm not sure we need to change original unless we understand why it was written like that.

Reply:
Just for reference: the original piece was added here: https://github.com/3scale/porta/pull/3351/files#diff-4c16dc8fb0816a60d4675d1987c5588b66341daf5edd80d0cb2fb1ececb5a9b6R47

This was to fix an issue where a reference was stored in our DB, but we couldn't find the `Stripe::Customer` by that ID anymore. So, previously the situation when the customer has been deleted was checked (and the fallback was to create a new customer). But the issue still occurred, when (if I remember correctly) switching the Stripe credentials - to point to a different Stripe instance. Obviously, in this case all old references become invalid.

I think in practical this changed code works almost the same way. The only caveat I see is that if `customer_id.blank?` and `create_customer` throws a `Stripe::InvalidRequestError` for some reason, then `create_customer` would be retried again (which was not happening originally, and it would just throw directly).

I can certainly revert this change to avoid confusion.